src/Lib/Components/Fills.vala

```diff
@@ -58,20 +106,50 @@ public class Akira.Lib.Components.Fills : Component, Copyable<Fills> {
         return cln;
     }
 
-    // Recommended accessors
+    // Recommended accessors.
+    public Fill? fill_from_id (int id) {
+        foreach (unowned var fill in data) {
+            if (fill.id == id) {
+                return fill.with_color (fill._color);
+            }
+        }
+        return null;
+    }
 
-    public Gee.ArrayList<Fill> fills () {
-        var tmp = new Gee.ArrayList<Fill> ();
+    public bool replace (Fill fill) {
         for (var i = 0; i < data.length; ++i) {
-            tmp.add (new Fill (data[i]));
+            if (data[i].id == fill.id) {
+                data[i] = fill;
+                return true;
+            }
         }
-        return tmp;
+        return false;
     }
 
-    public void prep_fills (uint number_to_prep) {
-        data = new Fill.FillData[number_to_prep];
-        for (var i = 0; i < number_to_prep; ++i) {
-            data[i]._id = i;
+    public void append (Fill fill) {
```

Comment:
don't add public mutators to components. This breaks the model contract and thus copy/paste/undo/redo

Reply:
What would you recommend to do here to handle multiple single `Fill` for the same `Fills` component?